src/backend/distributed/commands/table.c

```diff
@@ -186,8 +187,15 @@ PostprocessCreateTableStmt(CreateStmt *createStatement, const char *queryString)
 
 	if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
 	{
-		/* process CREATE TABLE ... PARTITION OF command */
-		PostprocessCreateTableStmtPartitionOf(createStatement, queryString);
+		/*
+		 * process CREATE TABLE ... PARTITION OF command
+		 * If that partition is already distributed, or if the table is not a partition,
+		 * just skip like in plain PG. In these cases we know that if_not_exists = true.
+		 */
+		if (!(IsCitusTable(relationId)) && PartitionTable(relationId))
+		{
+			PostprocessCreateTableStmtPartitionOf(createStatement, queryString);
```

Comment:
It still tries to distribute an existing table with the same name, which is a partition of some other table which is not distibuted. Because the existing table with the same name, satisfies these two conditions; it's not a citus table (not distributed) and it's a partition table. But errors out when distributing, because its parent relation is not distributed.

Example:
```
CREATE TABLE "test1" (
   "cdc" INTEGER,
   "product" INTEGER)
PARTITION BY LIST ("cdc");

CREATE TABLE "test2" (
   "cdc" INTEGER,
   "product" INTEGER)
PARTITION BY LIST ("cdc");

SELECT create_distributed_table('test1', 'product');

CREATE TABLE IF NOT EXISTS "test3" PARTITION OF "test2" FOR VALUES IN (1);

CREATE TABLE IF NOT EXISTS "test3" PARTITION OF "test1" FOR VALUES IN (1);

NOTICE:  relation "test3" already exists, skipping
ERROR:  cannot distribute relation "test3" which is partition of "test2"
DETAIL:  Citus does not support distributing partitions if their parent is not distributed table.
HINT:  Distribute the partitioned table "test2" instead.
```

Reply:
Makes sense, I will add a check for that